plugins/org.eclipse.n4js/src/org/eclipse/n4js/validation/validators/N4JSFunctionValidator.xtend

```diff
@@ -97,6 +108,49 @@ class N4JSFunctionValidator extends AbstractN4JSDeclarativeValidator {
 		// nop
 	}
 
+	/**
+	 * Checks all flow graph related validations
+	 */
+	@Check
+	def checkFlowGraphs(Script script) {
+		val ASTMetaInfoCache cache = astMetaInfoCacheHelper.getOrCreate(script.eResource() as N4JSResource);
+		val N4JSFlowAnalyses flowAnalyses = cache.getFlowAnalyses();
+		val dcf = new DeadCodeFinder();
+
+		flowAnalyses.analyze(dcf);
```

Comment:
remark about the API: I find it weird that method `#analyze()` isn't invoked on the `DeadCodeFinder`, here (but probably I haven't fully understood the meaning of N4JSFlowAnalyses yet)

Reply:
Why is `checkFlowGraphs` in the `N4JSFunctionValidator`??

Reply:
@mor-n4 : I renamed some methods and hope it is a little clearer now

@qtran-n4 : I replaced the old dead code finder in this class. Maybe it can be moved later, if appropriate.